File: cpp/i00.abstract/Triangle.cpp
#include "Triangle.h"

namespace ShapeSpace {

 /******* CONSTRUCTORS *******/
 Triangle::Triangle (double w, double h) : Shape(w,h) { /* nothing */ }

 /******* METHODS *******/
 double Triangle::calculateArea () { return (width * height / 2); }

}

File: javascripts/dart/h00.overriding/Triangle.dart
library Shape;
import "Shape.dart";

/******* CLASS DECLARATION *******/
class Triangle extends Shape {
  /******* CONSTRUCTORS *******/
  Triangle(double width, double height) : super(width, height) { /* Nothing */ }

  /******* METHODS *******/
  double calculateArea() { return (this.width * this.height)/2; }
}
